types/joi/joi-tests.ts

```diff
@@ -988,13 +988,13 @@ const Joi3 = Joi.extend({
         {
             name: 'asd',
             params: {
-                allowF: Joi.boolean().default(false),
+                allowFalse: Joi.boolean().default(false),
             },
             setup(params) {
-                const fIsAllowed = params.allowF;
+                const fIsAllowed = params.allowFalse;
             },
-            validate(params, value, state, options) {
-                if (value === 'asd' || params.allowF && value === 'asdf') {
+            validate(params, value: boolean, state, options) {
+                if (value || params.allowFalse && !value) {
```

Comment:
I think the existing test code was correct, if a bit confusing.

When defining a custom rule, "params" accepts an object mapping to Joi schemas. Each schema is used to validate the parameters passed to the rule. The value of "params" then becomes the validated object. In the original case, this will be an object with a property "allowF", which is a boolean defaulting to false.

The actual validation only allows the string values "asd", or in the case where the "allowF" parameter is true, "asdf".

The new validation you've written is also valid, but a future developer might read the test and think that the `Joi.boolean()` schema is related to `value`, which is not the case.